### PR TITLE
add STATE column to modules list

### PR DESCRIPTION
**Pull request by Cortey**
**Description**

Changes proposed in this pull request:

- add HEALTHY column

**Related issue(s)**

- #2264

### CODE REVIEW

File: internal/modules/list.go [hunk 1]
```diff
@@ -89,7 +98,169 @@ func List(ctx context.Context, client kyma.Interface) (ModulesList, error) {
...
+			}
+		}
+		if conditionUnwrapped["Type"] == "Warning" {
+			if conditionUnwrapped["status"] == "True" {
+				return "Warning"
+			}
+		}
+	}
+	return ""
+}
```
Comment (anoipm) [inline internal/modules/list.go:252]:
```suggestion
func getStateFromConditions(conditions []interface{}) string {
	for _, condition := range conditions {
		conditionUnwrapped := condition.(map[string]interface{})
		conditionStatus := conditionUnwrapped["status"]
		if conditionStatus != "True" {
			continue
		}
		conditionType := conditionUnwrapped["type"]
		switch conditionType {
			case "Available", "Processing", "Error", "Warning":
				return conditionType
		}
	}
	return ""
}
```

File: internal/modules/list.go [hunk 2]
```diff
@@ -89,7 +99,168 @@ func List(ctx context.Context, client kyma.Interface) (ModulesList, error) {
...
+	if kymaCR != nil {
+		for _, module := range kymaCR.Status.Modules {
+			if module.Name == moduleTemplate.Spec.ModuleName {
+				if module.State != "" {
+					return module.State
+				}
+			}
+		}
+	}
+	return ""
```
Comment (anoipm) [inline internal/modules/list.go:139]:
```suggestion
	if kymaCR == nil {
		return ""
	}
	for _, module := range kymaCR.Status.Modules {
		if module.Name == moduleTemplate.Spec.ModuleName && module.State != "" {
			return module.State
		}
	}
```

File: internal/modules/list.go [hunk 3]
```diff
@@ -89,7 +99,168 @@ func List(ctx context.Context, client kyma.Interface) (ModulesList, error) {
...
+
+	unstruct := generateUnstruct(apiVersion, kind, name, namespace)
+	result, err := client.RootlessDynamic().Get(ctx, &unstruct)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return "", err
+		} else {
+			return "", nil
+		}
+	}
```
Comment (anoipm) [inline internal/modules/list.go:164]:
```suggestion
	if errors.IsNotFound(err) {
		return "", nil
	}
	if err != nil {
		return "", err
	}
```

File: internal/modules/list.go [hunk 4]
```diff
@@ -89,7 +99,168 @@ func List(ctx context.Context, client kyma.Interface) (ModulesList, error) {
...
+
+	result, err := client.RootlessDynamic().Get(ctx, &unstruct)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return "", err
+		} else {
+			return "", nil
+		}
+
+	}
```
Comment (anoipm) [inline internal/modules/list.go:193]:
```suggestion
	if errors.IsNotFound(err) {
		return "", nil
	}
	if err != nil {
		return "", err
	}
```

File: internal/modules/list.go (hunk 2 above)
Comment (halamix2) [reply internal/modules/list.go:129]: I'll have to `return ""` anyway, this change would make code less readable

File: internal/modules/modules.go [hunk 5]
```diff
@@ -87,7 +100,114 @@ func List(ctx context.Context, client kyma.Interface) (ModulesList, error) {
 	return modulesList, nil
 }
 
-func getInstallDetails(kyma *kyma.Kyma, releaseMetas kyma.ModuleReleaseMetaList, moduleName string) ModuleInstallDetails {
+func getModuleDeploymentHealth(ctx context.Context, client kube.Client, moduleTemplate kyma.ModuleTemplate, kymaCR *kyma.Kyma) (Healthy, error) {
```
[the remark this answers is not in the document]
Comment (anoipm) [reply internal/modules/modules.go]: outdated

File: internal/modules/modules.go [hunk 6]
```diff
@@ -87,7 +100,114 @@ func List(ctx context.Context, client kyma.Interface) (ModulesList, error) {
...
+		}
+		if !errors.IsNotFound(err) {
+			return HealthyFalse, err
+		}
+	}
+
+	return HealthyUnknown, nil
+}
+
+func getResourceState(ctx context.Context, client kube.Client, apiVersion, kind, namespace, name string) (string, error) {
```
[the remark this answers is not in the document]
Comment (anoipm) [reply internal/modules/modules.go]: outdated

**Review by anoipm — APPROVED** [review]

**Pull request merged**